lib/node_runner_message_checker_test.go

```diff
@@ -0,0 +1,117 @@
+package sebak
+
+import (
+	"testing"
+
+	"github.com/stellar/go/keypair"
+	"github.com/stretchr/testify/assert"
+
+	"boscoin.io/sebak/lib/common"
+	"boscoin.io/sebak/lib/error"
+	"boscoin.io/sebak/lib/network"
+	"boscoin.io/sebak/lib/node"
+	"boscoin.io/sebak/lib/storage"
+)
+
+func MakeNodeRunner() (*NodeRunner, *sebaknode.LocalNode) {
+	kp, _ = keypair.Random()
+
+	nodeEndpoint := &sebakcommon.Endpoint{Scheme: "https", Host: "https://locahost:5000"}
+	localNode, _ := sebaknode.NewLocalNode(kp, nodeEndpoint, "")
+
+	vth, _ := NewDefaultVotingThresholdPolicy(66, 66)
+	is, _ := NewISAAC(networkID, localNode, vth)
+	st, _ := sebakstorage.NewTestMemoryLevelDBBackend()
+	network, _ := createNetMemoryNetwork()
+	nodeRunner, _ := NewNodeRunner(string(networkID), localNode, vth, network, is, st)
+	return nodeRunner, localNode
+}
+
+func TestMessageChecker(t *testing.T) {
+	var CheckerFuncs = []sebakcommon.CheckerFunc{
+		CheckNodeRunnerHandleMessageTransactionUnmarshal,
+		CheckNodeRunnerHandleMessageHasTransactionAlready,
+		CheckNodeRunnerHandleMessageHistory,
+		CheckNodeRunnerHandleMessagePushIntoTransactionPool,
+		CheckNodeRunnerHandleMessageTransactionBroadcast,
+	}
+
+	_, validTx := TestMakeTransaction(networkID, 1)
+	var b []byte
+	var err error
+
+	if b, err = validTx.Serialize(); err != nil {
+		return
+	}
+
+	validMessage := sebaknetwork.Message{Type: "message", Data: b}
+	nodeRunner, localNode := MakeNodeRunner()
+	checker := &NodeRunnerHandleMessageChecker{
+		DefaultChecker: sebakcommon.DefaultChecker{CheckerFuncs},
+		NodeRunner:     nodeRunner,
+		LocalNode:      localNode,
+		NetworkID:      networkID,
+		Message:        validMessage,
+	}
+
+	err = CheckNodeRunnerHandleMessageTransactionUnmarshal(checker)
+	assert.Nil(t, err)
+	assert.Equal(t, checker.Transaction, validTx)
+
+	err = CheckNodeRunnerHandleMessageHasTransactionAlready(checker)
+	assert.Nil(t, err)
+
+	err = CheckNodeRunnerHandleMessageHistory(checker)
+	assert.Nil(t, err)
+	var found bool
+	found, err = ExistsBlockTransactionHistory(checker.NodeRunner.Storage(), checker.Transaction.GetHash())
+	assert.True(t, found)
+
+	err = CheckNodeRunnerHandleMessagePushIntoTransactionPool(checker)
+	assert.Nil(t, err)
+	assert.True(t, checker.NodeRunner.Consensus().TransactionPool.Has(validTx.GetHash()))
+
+	// CheckNodeRunnerHandleMessageTransactionBroadcast(checker) is not suitable in unittest
```

Comment:
How about adding 'TODO'?

Reply:
Oh, this is not `TODO` thing. Just information for someone who think like this.
> Why `CheckNodeRunnerHandleMessageTransactionBroadcast function` is not checked?